google/compute_instance.go

```diff
@@ -0,0 +1,764 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func GetComputeInstanceCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	if obj, err := GetComputeInstanceApiObject(d, config); err == nil {
+		return Asset{
+			Name: fmt.Sprintf("//compute.googleapis.com/%s", obj["selfLink"]),
+			Type: "google.compute.Instance",
+			Resource: &AssetResource{
+				Version:              "v1",
+				DiscoveryDocumentURI: "https://www.googleapis.com/discovery/v1/apis/compute/v1/rest",
+				DiscoveryName:        "Instance",
+				Data:                 obj,
+			},
+		}, nil
+	} else {
+		return Asset{}, err
+	}
+}
+
+func GetComputeInstanceApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+	obj := make(map[string]interface{})
+	canIpForwardProp, err := expandComputeInstanceCanIpForward(d.Get("can_ip_forward"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("can_ip_forward"); !isEmptyValue(reflect.ValueOf(canIpForwardProp)) && (ok || !reflect.DeepEqual(v, canIpForwardProp)) {
+		obj["canIpForward"] = canIpForwardProp
+	}
+	disksProp, err := expandComputeInstanceDisks(d.Get("disks"), d, config)
```

Comment:
This is quite a bit more complicated in practice - needs to handle boot vs scratch disks vs attached disks.